tools/coqdoc/index.ml

```diff
@@ -240,7 +244,11 @@ let all_entries () =
     let l = try Hashtbl.find bt t with Not_found -> [] in
       Hashtbl.replace bt t ((s,m) :: l)
   in
-  let classify m (s,t) = (add_g s m t; add_bt t s m) in
+  let classify m (s,t) =
+    if incl_entry_type t then begin
+      add_g s m t; add_bt t s m
+    end
```

Comment:
Could you do this in an easier to understand way:
```suggestion
    if not (is_binder t) || !prefs.binder_index then begin
      add_g s m t; add_bt t s m
    end
```
where `is_binder` is the obvious function.

Reply:
@Alizter I consciously wrote `incl_entry_type` (now `include_entry` for more module naming consistency) as a separate function with maintainability in mind: it's likely people will want to control exactly what is included in their indexes in the future via `coqdoc` options. I disagree that replacing a straightforward separate match on `entry_type` with a logical expression that includes both a negation and a disjunction is an improvement for readability or understanding.

Reply:
OK that's fine then.